src/backend/distributed/master/master_delete_protocol.c

```diff
@@ -372,6 +376,12 @@ DropShards(Oid relationId, char *schemaName, char *relationName,
 
 	UseCoordinatedTransaction();
 
+	/*
+	 * We will use below variable accross this function to decide if we can
+	 * use local execution
+	 */
+	int32 localGroupId = GetLocalGroupId();
```

Comment:
maybe add the following to prevent any impression about this could happen on any node
```C
/* drop is currently only supported from the coordinator*/
Assert (localGroupId == COORDINATOR_GROUP_ID);
```

Reply:
thanks for this comment, applying it

Reply:
actually functions calling `DropShards` do already ensure it by calling `EnsureCoordinator`

Reply:
Yeah, but my point is in case someone else in some other code-path tries to call this one day. Plus, readability.  So, looks good now